packages/flutter/lib/src/material/text_field.dart

```diff
@@ -569,6 +569,11 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     assert(debugCheckHasDirectionality(context));
     final ThemeData themeData = Theme.of(context);
     final TextStyle style = widget.style ?? themeData.textTheme.subhead;
+
+    if (style.inherit == false && (style.fontSize == null || style.textBaseline == null)) {
+      throw FlutterError('Style must supply fontSize and textBaseline when inherit is false.');
+    }
```

Comment:
@HansMuller Is this a good place to do this validation? It seemed like I couldn't do it in the constructor since it's const, is that right?

Reply:
I think this check could be a TextField constructor assert. Something like:
```
assert(style == null || style.inherit || (style.fontSize != null && style.textBaseline != null), 'style must supply ...')
```

Reply:
I get errors when I try to access `style.something` in the initializers list, since TextField is a const constructor.  I also can't give the constructor a body for the same reason.  I don't fully understand the limitations on const constructors though, I'll play around with it and see if I can get something to work.

Reply:
Right, my bad. Method calls can't be "const evaluated". This assert will need to happen at build time.